polaris-core/src/main/java/org/apache/polaris/core/config/FeatureConfiguration.java

```diff
@@ -287,4 +294,25 @@ public static void enforceFeatureEnabledOrThrow(
                   + "This should only be set to 'true' for tests!")
           .defaultValue(false)
           .buildFeatureConfiguration();
+
+  public static final PolarisConfiguration<Integer> METADATA_CACHE_MAX_BYTES =
+      PolarisConfiguration.<Integer>builder()
+          .key("METADATA_CACHE_MAX_BYTES")
+          .catalogConfig("polaris.config.metadata-cache-max-bytes")
+          .description(
+              "If nonzero, the approximate max size a table's metadata can be in order to be cached in the persistence"
+                  + " layer. If zero, no metadata will be cached or served from the cache. If -1, all metadata"
+                  + " will be cached.")
+          .defaultValue(Constants.METADATA_CACHE_MAX_BYTES_NO_CACHING)
```

Comment:
Not a blocker: It's reasonable to turn it off by default. Do we have a recommended size so that users/admins could use it without additional evaluation work?